datastore/tasks.rb

```diff
@@ -0,0 +1,104 @@
+# Copyright 2016 Google, Inc
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in write, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# [START build_service]
+require "google/cloud"
+
+def create_client project_id:
+  gcloud = Google::Cloud.new project_id
+  gcloud.datastore
+end
+# [END build_service]
+
+# [START add_entity]
+def new_task project_id:, description:
+  datastore = create_client project_id: project_id
+
+  task = datastore.entity "Task" do |t|
+    t["description"] = description
+    t["created"] = Time.now
+    t["done"] = false
+    t.exclude_from_indexes! "description", true
+  end
+  datastore.save task
+
+  puts task.key.id
+  task.key.id
+end
+# [END add_entity]
+
+# [START update_entity]
+def mark_done project_id:, task_id:
+  datastore = create_client project_id: project_id
+
+  task = datastore.find "Task", task_id.to_i
+  task["done"] = true
+  datastore.save task
+end
+# [END update_entity]
+
+# [START retrieve_entities]
+def list_tasks project_id:
+  datastore = create_client project_id: project_id
+
+  query = datastore.query("Task").
```

Comment:
For the datastore queries, it seems like a pattern to put every .(something) on a new line
https://cloud.google.com/datastore/docs/concepts/queries#datastore-basic-query-ruby
there are more examples with 1 or 2 where, order, etc. on new lines